docs/jep/13062-bounded-while-loop.md

```diff
@@ -0,0 +1,121 @@
+# Bounded while loops
+
+- *Author: patrick-kidger (GitHub) / kidger (Google)*
+
+- *Date: November 2022*
+
+## Summary
+
+We propose to introduce "bounded while loops" into JAX: these are while loops which exit unconditionally after some maximum number of iterations (are "bounded"), and are reverse-mode autodifferentiable.
+
+An implementation is available at https://github.com/google/jax/pull/13062.
+
+## Background
+
+### JAX limitations
+
+In order to reverse-mode autodifferentiate an operation, then residuals must be stored between the forward and backward pass. In the case of a while loop, which may iterate for an arbitrary number of steps, then these residuals may be of potentially unbounded size. However, it is a limitation of JAX (inherited from XLA) that the shape of all arrays be known statically in advance. For this reason, `lax.while_loop` is not reverse-mode autodifferentiable (or to be precise, not transposable).
```

Comment:
In order to reverse-mode autodifferentiate an operation, the residuals must be stored between the forward and backward pass. In the case of a while loop, which may iterate for an arbitrary number of steps, then these residuals may be of potentially unbounded size. However, it is a limitation of JAX (inherited from XLA) that the shape of all arrays be known statically in advance. For this reason, `lax.while_loop` is not reverse-mode autodifferentiable (or to be precise, not transposable).